fieldservice/models/fsm_order.py

```diff
@@ -229,11 +245,77 @@ def _onchange_location_id_customer(self):
     equipment_id = fields.Many2one("fsm.equipment", string="Equipment")
 
     # Equipment used for all other Service Orders
-    equipment_ids = fields.Many2many("fsm.equipment", string="Equipments")
+    equipment_ids = fields.Many2many(
+        "fsm.equipment",
+        string="Equipments",
+        compute="_compute_equipment_ids",
+        precompute=True,
+        store=True,
+        readonly=False,
+    )
     type = fields.Many2one("fsm.order.type")
 
     internal_type = fields.Selection(related="type.internal_type")
 
+    @api.depends("company_id")
+    def _compute_equipment_ids(self):
+        for rec in self:
+            # Clear equipments that no longer match the order company
+            to_remove = rec.equipment_ids.filtered(
+                lambda equipment, rec=rec: equipment.company_id != rec.company_id
+            )
+            if to_remove:
+                rec.equipment_ids = [
+                    Command.unlink(equipment.id) for equipment in to_remove
+                ]
+            # If we have no equipments, auto populate if needed
+            if (
+                rec.company_id.auto_populate_equipments_on_order
+                and not rec.equipment_ids
+            ):
+                rec.equipment_ids = self.env["fsm.equipment"].search(
+                    [
+                        ("current_location_id", "=", rec.location_id.id),
+                        ("company_id", "=", rec.company_id.id),
+                    ]
+                )
+
+    @api.depends("location_id")
+    def _compute_location_directions(self):
+        for rec in self:
+            rec.location_directions = rec.location_id.complete_direction
+
+    @api.depends("template_id")
+    def _compute_todo(self):
+        for rec in self:
+            if rec.template_id:
+                rec.todo = rec.template_id.instructions
+
+    @api.depends("equipment_ids", "equipment_id", "type")
+    def _compute_description(self):
+        for rec in self:
+            if rec.description:
+                continue
+            equipments = (
+                rec.equipment_ids
+                if rec.type and rec.internal_type not in ("repair", "maintenance")
+                else rec.equipment_id
+            )
+            rec.description = "\n".join(
+                equipment.notes for equipment in equipments if equipment.notes
+            )
+
+    @api.depends("territory_id")
+    def _compute_person_id(self):
+        """Compute the person from the territory"""
+        for rec in self:
+            # If the person is one of the territory's workers, keep it.
+            if rec.person_id in rec.territory_id.person_ids:
```

Comment:
Does `res.territory` model have both fields: `person_ids` and `person_id`? Just checking, because IDK

Reply:
It does :)

https://github.com/OCA/field-service/blob/b6ff55995cf376cd5e64f9301f63a6a2edb2bae4/fieldservice/models/res_territory.py#L7-L11